File: command/SmartHomeShortcut.java
package command;

import java.util.HashMap;
import java.util.Map;

public class SmartHomeShortcut {

    private final Map<String, Command> commands;

    public SmartHomeShortcut() {
        this.commands = new HashMap<>();
    }

    public void setCommand(String shortcutCommand, Command command) {
        commands.put(shortcutCommand, command);
    }

    public void openShortcut(String shortcutCommand) {
        commands.get(shortcutCommand).execute();
    }
}

File: typescript/patterns/command/smart-home-shortcut.ts
class SmartHomeShortcut {
  private commands: Map<string, Command>;
  constructor() {
    this.commands = new Map<string, Command>();
  }

  openShortcut(shortcut: string): void {
    this.commands.get(shortcut)?.execute();
  }

  setCommand(shortcutCommand: string, command: Command): void {
    this.commands.set(shortcutCommand, command);
  }
}
